compaction/compaction_strategy.cc

```diff
@@ -283,37 +284,46 @@ class time_window_backlog_tracker final : public compaction_backlog_tracker::imp
         return b;
     }
 
-    // FIXME: Should provide strong exception safety guarantees
+    // Provides strong exception safety guarantees
     virtual void replace_sstables(const std::vector<sstables::shared_sstable>& old_ssts, const std::vector<sstables::shared_sstable>& new_ssts) override {
         struct replacement {
             std::vector<sstables::shared_sstable> old_ssts;
             std::vector<sstables::shared_sstable> new_ssts;
         };
         std::unordered_map<api::timestamp_type, replacement> per_window_replacement;
+        auto tmp_windows = _windows;
 
```

Comment:
Could be made less stall-y by only copying affected windows, then only moving back affected windows.

But can be deferred.

Reply:
Isn't moving back a subset of the windows unsafe?